File: Moodmate.php
<?php
$moodData = [
    "happy" => [
        "quote" => "Happiness is not by chance, but by choice.",
        "song" => "Pharrell - Happy",
        "food" => "Ice Cream"
    ],
    "sad" => [
        "quote" => "Tears come from the heart, not the brain.",
        "song" => "Adele - Someone Like You",
        "food" => "Dark Chocolate"
    ],
    "stressed" => [
        "quote" => "Just breathe. You are strong.",
        "song" => "Lo-fi Chill Mix",
        "food" => "Herbal Tea"
    ],
    "angry" => [
        "quote" => "Speak when you are calm, not when you are burning.",
        "song" => "Eminem - Lose Yourself",
        "food" => "Grilled Chicken"
    ],
    "romantic" => [
        "quote" => "Love is composed of a single soul inhabiting two bodies.",
        "song" => "Ed Sheeran - Perfect",
        "food" => "Strawberries & Wine"
    ],
    "neutral" => [
        "quote" => "Balance is the key to everything.",
        "song" => "Lo-fi Beats",
        "food" => "Fresh Salad"
    ]
];

function logMood($mood) {
    $timestamp = date('Y-m-d H:i:s');
    $logEntry = "$timestamp - Mood: $mood\n";
    file_put_contents('mood_log.txt', $logEntry, FILE_APPEND);
}

function prompt($prompt_msg){
    echo $prompt_msg;
    return trim(fgets(STDIN));
}

echo "Welcome to MoodMate - The Emotional Console\n";

$moods = array_keys($moodData);
foreach ($moods as $index => $mood) {
    echo ($index + 1) . ". " . ucfirst($mood) . "\n";
}

$choice = prompt("Enter choice (1-" . count($moods) . "): ");

if (!is_numeric($choice) || $choice < 1 || $choice > count($moods)) {
    echo "Invalid choice. Exiting.\n";
    exit(1);
}

$selectedMood = $moods[$choice - 1];
$info = $moodData[$selectedMood];

echo "\nQuote: \"{$info['quote']}\"\n";
echo "Suggested Song: {$info['song']}\n";
echo "Suggested Food: {$info['food']}\n";

logMood($selectedMood);

echo "\nYour mood has been logged. Take care!\n";
?>

File: MoodMate.cs
using System;
using System.Collections.Generic;
using System.IO;

class MoodInfo
{
    public string Quote { get; set; }
    public string Song { get; set; }
    public string Food { get; set; }

    public MoodInfo(string quote, string song, string food)
    {
        Quote = quote;
        Song = song;
        Food = food;
    }
}

class MoodMate
{
    static Dictionary<string, MoodInfo> moodData = new Dictionary<string, MoodInfo>
    {
        { "happy", new MoodInfo(
            "Happiness is not by chance, but by choice.",
            "Pharrell - Happy",
            "Ice Cream") },

        { "sad", new MoodInfo(
            "Tears come from the heart, not the brain.",
            "Adele - Someone Like You",
            "Dark Chocolate") },

        { "stressed", new MoodInfo(
            "Just breathe. You are strong.",
            "Lo-fi Chill Mix",
            "Herbal Tea") },

        { "angry", new MoodInfo(
            "Speak when you are calm, not when you are burning.",
            "Eminem - Lose Yourself",
            "Grilled Chicken") },

        { "romantic", new MoodInfo(
            "Love is composed of a single soul inhabiting two bodies.",
            "Ed Sheeran - Perfect",
            "Strawberries & Wine") },

        { "neutral", new MoodInfo(
            "Balance is the key to everything.",
            "Lo-fi Beats",
            "Fresh Salad") }
    };

    static void LogMood(string mood)
    {
        string timestamp = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss");
        string entry = $"{timestamp} - Mood: {mood}\n";
        File.AppendAllText("mood_log.txt", entry);
    }

    static void Main()
    {
        Console.WriteLine("Welcome to MoodMate - The Emotional Console");
        var moodKeys = new List<string>(moodData.Keys);

        for (int i = 0; i < moodKeys.Count; i++)
        {
            Console.WriteLine($"{i + 1}. {CultureInfo.CurrentCulture.TextInfo.ToTitleCase(moodKeys[i])}");
        }

        Console.Write($"Enter choice (1-{moodKeys.Count}): ");
        if (!int.TryParse(Console.ReadLine(), out int choice) || choice < 1 || choice > moodKeys.Count)
        {
            Console.WriteLine("Invalid choice. Exiting.");
            return;
        }

        string selectedMood = moodKeys[choice - 1];
        MoodInfo info = moodData[selectedMood];

        Console.WriteLine($"\nQuote: \"{info.Quote}\"");
        Console.WriteLine($"Suggested Song: {info.Song}");
        Console.WriteLine($"Suggested Food: {info.Food}");

        LogMood(selectedMood);

        Console.WriteLine("\nYour mood has been logged. Take care!");
    }
}
